lib/java/src/org/apache/thrift/async/TAsyncMethodCall.java

```diff
@@ -225,8 +231,13 @@ private void cleanUpAndFireCallback(SelectionKey key) {
     key.interestOps(0);
     // this ensures that the TAsyncMethod instance doesn't hang around
     key.attach(null);
```

Comment:
I started working on another async issue (THRIFT-3680) on top of your change.
As I actually ran your code, I think it immediately became evident that the try-catch is simply for new checked exception, period.
In that case, I'd like to put cancel() to the **catch** part to restore the original behavior.
Any concern for that ?